src/WaCore.Common/ExceptionHandling/WacHandlingConfiguration.cs

```diff
@@ -0,0 +1,237 @@
+                    {
+                        handler(castedException, i);
+                        return new HandlingResult<TResult>(true);
+                    }
+                }
+
+                return new HandlingResult<TResult>(false);
+            });
+
+            return this;
+        }
+
+        public WacHandlingConfiguration<TParameter, TResult> On<TException>(
+            Func<TException, HandlingInput<TParameter>, HandlingResult<TResult>> handler,
+            Func<TException, HandlingInput<TParameter>, bool> condition = null)
+            where TException : Exception
+        {
+            if (handler == null) throw new ArgumentNullException(nameof(handler));
+
+            AddHandler((ex, i) =>
+            {
+                if (TryCastException(ex, out TException castedException))
+                {
+                    if (condition == null || condition(castedException, i))
+                    {
+                        return handler(castedException, i);
+                    }
+                }
+
+                return new HandlingResult<TResult>(false);
+            });
+
+            return this;
+        }
+
+        public WacHandlingConfiguration<TParameter, TResult> On<TException>(
+            Action<TException> handler,
+            Func<TException, HandlingInput<TParameter>, bool> condition = null)
+            where TException : Exception
+        {
+            if (handler == null) throw new ArgumentNullException(nameof(handler));
+
+            AddHandler((ex, i) =>
+            {
+                if (TryCastException(ex, out TException castedException))
+                {
+                    if (condition == null || condition(castedException, i))
+                    {
+                        handler(castedException);
+                        return new HandlingResult<TResult>(true);
+                    }
+                }
+
+                return new HandlingResult<TResult>(false);
+            });
+
+            return this;
+        }
+
+        public WacHandlingConfiguration<TParameter, TResult> On<TException>(
+            Func<TException, HandlingResult<TResult>> handler,
+            Func<TException, HandlingInput<TParameter>, bool> condition = null)
+            where TException : Exception
+        {
+            if (handler == null) throw new ArgumentNullException(nameof(handler));
+
+            AddHandler((ex, i) =>
+            {
+                if (TryCastException(ex, out TException castedException))
+                {
+                    if (condition == null || condition(castedException, i))
+                    {
+                        return handler(castedException);
+                    }
+                }
+
+                return new HandlingResult<TResult>(false);
+            });
+
+            return this;
+        }
+
+        public WacHandlingConfiguration<TParameter, TResult> FinalizeWith(
```

Comment:
I was wondering where you would implement Logging of exceptions.
Is `FinalizeWith` the method, where you would do that?
If not, what is an example, where this method would be used?